packages/@aws-cdk/aws-location-alpha/lib/map.ts

```diff
@@ -0,0 +1,321 @@
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { ArnFormat, IResource, Lazy, Resource, Stack, Token } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+import { CfnMap } from 'aws-cdk-lib/aws-location';
+import { generateUniqueId } from './util';
+
+export interface IMap extends IResource {
+  /**
+   * The name of the map
+   *
+   * @attribute
+   */
+  readonly mapName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the Map
+   *
+   * @attribute Arn,MapArn
+   */
+  readonly mapArn: string;
+}
+
+/**
+ * Properties for a map
+ */
+export interface MapProps {
+  /**
+   * A name for the map
+   *
+   * @default - A name is automatically generated
+   */
+  readonly mapName?: string;
+
+  /**
+   * A description for the map
+   *
+   * @default - no description
+   */
+  readonly description?: string;
+
+  /**
+   * Specifies the map style selected from an available data provider.
+   */
+  readonly style: Style;
+
+  /**
+   * Specifies the custom layers for the style.
+   *
+   * @default - no cutom layes
+   */
+  readonly customLayers?: CustomLayer[];
+
+  /**
+   * Specifies the map political view selected from an available data provider.
+   *
+   * The political view must be used in compliance with applicable laws, including those laws about mapping of the country or region where the maps,
+   * images, and other data and third-party content which you access through Amazon Location Service is made available.
+   *
+   * @see https://docs.aws.amazon.com/location/latest/developerguide/map-concepts.html#political-views
+   *
+   * @default - no political view
+   */
+  readonly politicalView?: PoliticalView;
+}
+
+/**
+ * An additional layer you can enable for a map style.
+ */
+export enum CustomLayer {
+  /**
+   * The POI custom layer adds a richer set of places, such as shops, services, restaurants, attractions, and other points of interest to your map.
+   * Currently only the VectorEsriNavigation map style supports the POI custom layer.
```

Comment:
I considered adding this validation, but decided against it as I thought the scope might expand in the future, potentially making it unnecessary. Please provide any opinions if you have them.